contracts/MerkleDrop.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "hardhat/console.sol";
+import "./IDrop.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MerkleDrop is IDrop, Ownable {
+    using SafeERC20 for IERC20;
+
+    address public override token;
+    bytes32 public override merkleRoot;
+    bool public initialized;
+    uint256 public expireTimestamp;
+
+    mapping(address => bool) public claimed;
+
+    function init(
+        address owner_,
+        address token_,
+        bytes32 merkleRoot_,
+        uint256 expireTimestamp_
+    ) external {
+        require(!initialized, "Drop already Initialized");
+        initialized = true;
+        token = token_;
+        merkleRoot = merkleRoot_;
+        expireTimestamp = expireTimestamp_;
+        _transferOwnership(owner_);
+    }
+
+    function claim(
+        address account,
+        uint256 amount,
+        bytes32[] calldata merkleProof
+    ) external override {
+        require(!claimed[account], "Drop already claimed.");
+        bytes32 node = keccak256(abi.encodePacked(account, amount));
+        require(
+            MerkleProof.verify(merkleProof, merkleRoot, node),
```

Comment:
The last parameter is meant to be `leaf` or `node`?

Reply:
we can call it the leaf of a tree or node of the tree. I'll change it to leaf 😄